src/x-platform/index.js

```diff
@@ -24,42 +25,56 @@ export function onElectron(event, callback) {
   return false
 }
 
-export function notificate({title, content, icon, slug}, callback = noop) {
-  if (macGap) {
-    macGap.notify({
-      title,
-      content,
-      sound: false
-    })
-    return
+export class Notifier {
```

Comment:
if you created a class then it can be exactly the w3c interface .... I thought we don't need a class, thats why I suggested createNotification ... in this case it can be new Notification()

https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API/Using_the_Notifications_API